doc/Uplink.xml

```diff
@@ -359,10 +359,11 @@
           <para>Error : Optional user readable error information (readonly).</para>
         </listitem>
       </itemizedlist>
-      <para>The device shall interpret the field RemoteAddress as key to a specific uplink
-        configuration.</para>
-      <para>A device shall reject a configuration containing both CertificateID and
-        AuthorizationServer. </para>
+      <para>A device shall interpret the field RemoteAddress as key to a specific uplink
+        configuration and it shall support configuring the opional parameters CertificateID and
+        CertPathValidationPolicyID.</para>
```

Comment:
Isn't this only needed if mTLS is mentioned in AuthorizationModes?

Reply:
Uplink spec initially supports mTLS authentication before jwt is proposed.

Reply:
True, but after the addition of JWT is an implementer mandated to always support mTLS or only if it is signaled via AuthorizationModes?

Reply:
As I remember mTLS Authentication i.e certificated based mutual tls authentication b/w device and cloud service uplink endpoint is independent of Authorization server configuration types and Client authentication methods.

Reply:
Here is what I suggest should reflect in spec to avoid any confusion 

  - **CertificateID** field is mandatory when Authorization mode is mTLS
  - **AuthorizationServer** Configuration is mandatory when Authorization mode is JWT
  - **CertPathValidationPolicyID** is mandatory for either Authorization modes mTLS/JWT as per the requirement in section 5.3. **The remote client shall authenticate itself using a valid server certificate. The local service shall verify the validity of the remote certificate according to RFC 6125.**
    - Since this field is already specified as optional - let's say if client does not configure anything don't we need to have some kind of fallback mechanism to use default/device in-built trust anchors?  otherwise the above requirement of remote certificate validation cannot be met. 
    
@jmelancongen  comments?

Reply:
> Yep that fits my understanding of the fields today. The exception being CertPathValidationPolicyID being optional, where section 6.1 states that the certificate is not validated if this is not provided (although I hate it very much, as falling back to "no security" feels quite wrong)

Can we clarify that requirement as below?
  - **If not configured, server certificate validation is left to the device implementation, for e.g. device can choose to validate certificate using default validation policy or choose to not do the validation.**
  
 This kind of clarification can still maintain backward compatibility with existing implementations?

Reply:
I personally don't mind, although worded like that, it pretty much means for clients that they must set a policy, otherwise they don't really know what a device will do.
From a security standpoint, that's not really a problem, because the previous wording is completely insecure anyway so we had to configure something in any case

Reply:
unless we bring in something new that device can signal in say SetUplink response to confirm that it is using fall back mechanism for validation? But may be it is too late by then and client can't do much with such information other than feeling a bit more secure :-)

Reply:
> I personally don't mind, although worded like that, it pretty much means for clients that they must set a policy, otherwise they don't really know what a device will do. From a security standpoint, that's not really a problem, because the previous wording is completely insecure anyway so we had to configure something in any case

Added https://github.com/onvif/specs/pull/481 to discuss separately.

Reply:
I am in agreement with @jmelancongen on this and pretty much agree with your suggestion.